crates/edr_napi/src/solidity_tests.rs

```diff
@@ -12,95 +12,67 @@ use napi::{
     },
     tokio,
     tokio::runtime,
-    Env, JsFunction,
+    JsFunction,
 };
 use napi_derive::napi;
 
 use crate::solidity_tests::{
     runner::build_runner, test_results::SuiteResult, test_suite::TestSuite,
 };
 
-/// Executes solidity tests.
-#[napi]
-pub struct SolidityTestRunner {
-    /// The callback to call with the results as they become available.
-    results_callback_fn: ThreadsafeFunction<SuiteResult>,
+/// Executes Solidity tests.
+///
+/// The function will return as soon as test execution is started.
+/// The progress callback will be called with the results of each test suite.
+/// It is up to the caller to track how many times the callback is called to
+/// know when all tests are done.
+// False positive from Clippy. The function is exposed through the FFI.
+#[allow(dead_code)]
+#[napi(
+    ts_args_type = "test_suites: Array<TestSuite>, gas_report: boolean, progress_callback: (result: SuiteResult) => void"
```

Comment:
You can specify [`ts_arg_type`](https://napi.rs/docs/concepts/types-overwrite.en#ts_arg_type) for just a single parameter of the function above `progress_callback`

Reply:
Thanks, fixed